scripts/update.py

```diff
@@ -0,0 +1,453 @@
+    ]
+
+    try:
+        ret = subprocess.run(cmd,
+                             stdin=subprocess.DEVNULL,
+                             capture_output=True,
+                             cwd=CWD,
+                             timeout=60.0)
+        if ret.returncode != 0:
+            print(ret.stderr.decode())
+            return False
+        with open("comptime.{}.revision".format(RUNNER_ID), "w") as f:
+            f.write(LLVM_REV)
+        return True
+
+    except Exception as e:
+        print("Internal error getting comptime diff: {}".format(str(e)))
+        return False
+
+
+def get_current_llvm_rev():
+    return run_cmd("git -C llvm/llvm-project rev-parse HEAD")
+
+
+def build_llvm():
+    global CWD
+    global STATS_OUT
+    global COMPTIME_OUT
+    global OPT_EXEC
+
+    cmd = [
+        "cmake", "-S", "llvm/llvm-project/llvm", "-DCMAKE_BUILD_TYPE=Release",
+        "-G", "Ninja", "-DLLVM_INCLUDE_EXAMPLES=OFF",
+        "-DLLVM_OPTIMIZED_TABLEGEN=ON", "-DLLVM_ENABLE_WARNINGS=OFF",
+        "-DLLVM_APPEND_VC_REV=OFF", "-DLLVM_TARGETS_TO_BUILD='X86'",
+        "-DCMAKE_C_COMPILER_LAUNCHER=ccache",
+        "-DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
+    ]
+    build_dir = "llvm/llvm-build"
+    if STATS_OUT:
+        cmd.append("-DLLVM_FORCE_ENABLE_STATS=ON")
+        build_dir += "-stats"
+    if COMPTIME_OUT:
+        build_dir += "-comptime"
+    else:
+        cmd.append("-DLLVM_ENABLE_ASSERTIONS=ON")
+
+    cmd.append("-B")
+    cmd.append(build_dir)
+
+    try:
+
+        ret = subprocess.run(cmd,
+                             stdin=subprocess.DEVNULL,
+                             capture_output=True,
+                             cwd=CWD,
+                             timeout=6000.0)
+
+        if ret.returncode != 0:
+            print(ret.stderr.decode())
+            return False
+
+        cmd = ["cmake", "--build", build_dir, "-j", "-t", "opt"]
+        ret = subprocess.run(cmd,
+                             stdin=subprocess.DEVNULL,
+                             capture_output=True,
+                             cwd=CWD,
+                             timeout=6000.0)
+
+        if ret.returncode != 0:
+            print(ret.stderr.decode())
+            return False
+
+        opt = os.path.join(build_dir, "bin", "opt")
+        assert os.path.isfile(opt) and os.access(opt, os.X_OK)
+        OPT_EXEC = opt
+        return True
+    except Exception as e:
+        print("Internal error building llvm: {}".format(str(e)))
+        return False
+
+
+def run_opt(task):
+    global OPT_EXEC
+    global COMPTIME_OUT
+    global STATS_OUT
+    global CWD
+
+    assert OPT_EXEC is not None
+    input_file, output_file = task
+    try:
+        cmd = [
+            OPT_EXEC, "-O3", "-disable-loop-unrolling",
+            "-vectorize-loops=false", "-vectorize-slp=false", input_file, "-S"
+        ]
+        tmp_output = output_file + ".bench_tmp.ll"
+        cmd += ["-o", tmp_output]
```

Comment:
```suggestion
        cmd += ["-o", tmp_output] if COMPTIME_OUT is None else ['--disable-output']
```
It would make the instruction count more stable.